data-contracts/README.md

```diff
@@ -0,0 +1,439 @@
+
+```avroschema
+{
+  "name": "Membership",
+  "namespace": "io.confluent.devrel",
+  "type": "record",
+  "fields": [
+    {"name": "user_id", "type": "string"},
+    {"name": "start_date", "type": {"type": "int", "logicalType": "date"}},
+    {"name": "end_date", "type": {"type": "int", "logicalType": "date"}}
+  ]
+}
+```
+
+Using the [schema-registry-plugin](https://github.com/ImFlog/schema-registry-plugin) for gradle, we can register this schema with the Confluent Schema Registry.
+
+```kotlin
+schemaRegistry {
+    ...
+    register {
+        subject(inputSubject =  "membership-avro-value", type = "AVRO", file = "$avroSchemaDir/membership_v1.avsc")
+            .setMetadata("$metadataDir/membership_major_version_1.json")
+    }
+}    
+```
+
+Remember the `compatibility_group` parameter from the subject configuration? This now comes into play as we set the metadata on the schema to denote this version as `major_version=1`:
+
+```json
+{
+  "properties": {
+    "major_version": 1
+  }
+}
+```
+
+Schema registration is completed via the `registerSchemas` task of the `schemas` module:
+
+```shell
+./gradlew :data-contracts:schemas:registerSchemasTask
+```
+
+To use this schema in an application - `app-schema-v1` - we utilize the same gradle plugin's `downloadSchemasTask`. First we define the subjects we want to download. In this example, we copy the schema to directory `src/main/avro`:
+
+```kotlin
+schemaRegistry {
+    download {
+        // download the membership avro schema, version 1
+        subject("membership-avro-value", "${projectDir}/src/main/avro", 1)
+    }
+}
+```
+
+With the schema(s) downloaded, we next want to generate Java code to provide our application an SDK with compile-time bindings to serialize and deserialize events to and from the `membership-avro` topic. Let's use the [gradle-avro-dependency-plugin](https://github.com/bakdata/gradle-avro-dependency-plugin) to generate code, using the `generateAvroJava` task.
+
+For simplicity, I have encapsulated `downloadSchemasTask` with the `generateAvroJava` task into a custom gradle task named `generateCode`, executed as follows:
+
+```shell
+./gradlew :data-contracts:app-schema-v1:generateCode
+```
+
+Now implement classes to produce and consume events with this schema. First the producer class, extending `BaseProducer`:
+
+```kotlin
+class MembershipProducer: BaseProducer<String, Membership>(mapOf(
+    ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG to "org.apache.kafka.common.serialization.StringSerializer",
+    ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG to "io.confluent.kafka.serializers.KafkaAvroSerializer",
+    ProducerConfig.CLIENT_ID_CONFIG to "membership-producer-app-v1",
+    AbstractKafkaSchemaSerDeConfig.LATEST_COMPATIBILITY_STRICT to true,
+    AbstractKafkaSchemaSerDeConfig.USE_LATEST_VERSION to false,
+    AbstractKafkaSchemaSerDeConfig.USE_LATEST_WITH_METADATA to "major_version=1"
+))
```

Comment:
i think the code snippets can go away, or be trimmed down to the most important part - last 3 lines

Reply:
@davetroiano Do you think I should remove the references in the paragraph below to the Serializers?

Reply:
I could go either way on that... it doesn't seem necessary to cover the tutorial topic but it's close enough to the topic that I think it's alright to leave in there